### PR TITLE
[MSHARED-685] assert some things

### PR DESCRIPTION
**Pull request by elharo**
@michael-o also fixes a couple of FindBugs warnings

### CODE REVIEW

**Comment by elharo** [comment]
See https://builds.apache.org/job/maven-box/job/maven-shared-utils/job/i848/3/

Changes:

* restore asserts

5 tests failed.
FAILED:  org.apache.maven.shared.utils.io.DirectoryScannerTest.windows-jdk14 / Build windows-jdk14 / checkSymlinkBehaviour

Error Message:
null

Stack Trace:
java.lang.AssertionError
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.assertAlwaysIncluded(DirectoryScannerTest.java:172)
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.checkSymlinkBehaviour(DirectoryScannerTest.java:140)


FAILED:  org.apache.maven.shared.utils.io.DirectoryScannerTest.windows-jdk11 / Build windows-jdk11 / checkSymlinkBehaviour

Error Message:
null

Stack Trace:
java.lang.AssertionError
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.assertAlwaysIncluded(DirectoryScannerTest.java:172)
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.checkSymlinkBehaviour(DirectoryScannerTest.java:140)


FAILED:  org.apache.maven.shared.utils.io.DirectoryScannerTest.windows-jdk8 / Build windows-jdk8 / checkSymlinkBehaviour

Error Message:
null

Stack Trace:
java.lang.AssertionError
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.assertAlwaysIncluded(DirectoryScannerTest.java:172)
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.checkSymlinkBehaviour(DirectoryScannerTest.java:140)


FAILED:  org.apache.maven.shared.utils.io.DirectoryScannerTest.windows-jdk7 / Build windows-jdk7 / checkSymlinkBehaviour

Error Message:
null

Stack Trace:
java.lang.AssertionError
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.assertAlwaysIncluded(DirectoryScannerTest.java:172)
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.checkSymlinkBehaviour(DirectoryScannerTest.java:140)


FAILED:  org.apache.maven.shared.utils.io.DirectoryScannerTest.windows-jdk13 / Build windows-jdk13 / checkSymlinkBehaviour

Error Message:
null

Stack Trace:
java.lang.AssertionError
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.assertAlwaysIncluded(DirectoryScannerTest.java:172)
        at org.apache.maven.shared.utils.io.DirectoryScannerTest.checkSymlinkBehaviour(DirectoryScannerTest.java:140)

File: src/test/java/org/apache/maven/shared/utils/io/DirectoryScannerTest.java [hunk 1]
```diff
@@ -50,7 +50,10 @@ private void createTestData()
     {
         File rootDir = tempFolder.getRoot();
         File folder1 = new File( rootDir, "folder1" );
-        folder1.mkdirs();
+        if (!folder1.mkdirs())
```
Comment (michael-o) [inline src/test/java/org/apache/maven/shared/utils/io/DirectoryScannerTest.java]: spaces please

File: src/test/java/org/apache/maven/shared/utils/io/DirectoryScannerTest.java [hunk 2]
```diff
@@ -60,7 +63,10 @@ private void createTestData()
         FileTestHelper.generateTestFile( new File( folder1, "file5.dat" ), 15 );
 
         File folder2 = new File( folder1, "ignorefolder" );
-        folder2.mkdirs();
+        if (!folder2.mkdirs())
```
Comment (michael-o) [inline src/test/java/org/apache/maven/shared/utils/io/DirectoryScannerTest.java]: same here

File: src/test/java/org/apache/maven/shared/utils/io/DirectoryScannerTest.java (hunk 1 above)
Comment (elharo) [reply src/test/java/org/apache/maven/shared/utils/io/DirectoryScannerTest.java]: done

File: src/test/java/org/apache/maven/shared/utils/io/DirectoryScannerTest.java (hunk 2 above)
Comment (elharo) [reply src/test/java/org/apache/maven/shared/utils/io/DirectoryScannerTest.java]: done

**Comment by jira-importer** [comment]
Resolve #263

**Pull request merged**